pretty_gpx/common/data/overpass_processing.py

```diff
@@ -1,7 +1,6 @@
 #!/usr/bin/python3
```

Comment:
Sorry @JulesL2, the last weeks have been quite busy and i struggled to find some spare time for working on pretty-gpx

I'll do my best to review it in the next days :crossed_fingers: 

Can you help me reviewing by providing some factual data? Many thanks and have a nice week-end


I'd like to have a working release that we can share for Xmas. With the bridge and Points of interest as well :) 
Do you have in mind to print one of your race for xmas ?

Reply:
Many thanks again for your time :fire:

Reply:
@ThomasParistech No problem, this is just a hobby no place for pressure and stress !

I will provide you examples of improvemement. 
To be short, before you could apply several time merge ways and you would still merge way because 1, 2 or 3 pass where not enough to merge all ways that are mergeable as the function did not take all case into account start+end and mix

Now when you try to merge a second time after a first merge you merge 0 way because everything that was mergeable has been merge

Yes, printing maps for christmas would be great :)

Reply:
In short, for lakes and other stuff this update reduce the occurence of surface that are not well drawn when the line was uncomplete to define the surface due to missing merges (sometimes rivers surfaces looked bad)

Reply:
@ThomasParistech 
The following test is : 
In simplify_ways function instead of calling 1 time the merge_ways, we call 3 times in a row:

With marathon_paris.gpx file:

Before
```
Highways:
[ 2024/12/07 - 17:30:52 | pretty-gpx | INFO ] 7750 segments merged
[ 2024/12/07 - 17:30:52 | pretty-gpx | INFO ] 76 segments merged
[ 2024/12/07 - 17:30:52 | pretty-gpx | INFO ] 34 segments merged
Secondary roads:
[ 2024/12/07 - 17:30:53 | pretty-gpx | INFO ] 9374 segments merged
[ 2024/12/07 - 17:30:53 | pretty-gpx | INFO ] 197 segments merged
[ 2024/12/07 - 17:30:53 | pretty-gpx | INFO ] 84 segments merged
Residential roads:
[ 2024/12/07 - 17:30:55 | pretty-gpx | INFO ] 13360 segments merged
[ 2024/12/07 - 17:30:55 | pretty-gpx | INFO ] 438 segments merged
[ 2024/12/07 - 17:30:56 | pretty-gpx | INFO ] 100 segments merged
Service roads:
[ 2024/12/07 - 17:31:00 | pretty-gpx | INFO ] 8899 segments merged
[ 2024/12/07 - 17:31:01 | pretty-gpx | INFO ] 271 segments merged
[ 2024/12/07 - 17:31:01 | pretty-gpx | INFO ] 46 segments merged
```



After: 
```
Highways:
[ 2024/12/07 - 17:23:00 | pretty-gpx | INFO ] 8438 segments merged
[ 2024/12/07 - 17:23:00 | pretty-gpx | INFO ] 0 segments merged
[ 2024/12/07 - 17:23:00 | pretty-gpx | INFO ] 0 segments merged
[ 2024/12/07 - 17:23:00 | pretty-gpx | INFO ] Hausdorff distance simplified 4.09e+03m
Secondary roads:
[ 2024/12/07 - 17:23:01 | pretty-gpx | INFO ] 10499 segments merged
[ 2024/12/07 - 17:23:01 | pretty-gpx | INFO ] 0 segments merged
[ 2024/12/07 - 17:23:01 | pretty-gpx | INFO ] 0 segments merged
[ 2024/12/07 - 17:23:01 | pretty-gpx | INFO ] Hausdorff distance simplified 6.53e+03m
Residential roads:
[ 2024/12/07 - 17:23:01 | pretty-gpx | INFO ] 15854 segments merged
[ 2024/12/07 - 17:23:02 | pretty-gpx | INFO ] 0 segments merged
[ 2024/12/07 - 17:23:02 | pretty-gpx | INFO ] 0 segments merged
[ 2024/12/07 - 17:23:03 | pretty-gpx | INFO ] Hausdorff distance simplified 2.71e+04m
Service roads:
[ 2024/12/07 - 17:23:04 | pretty-gpx | INFO ] 12281 segments merged
[ 2024/12/07 - 17:23:04 | pretty-gpx | INFO ] 0 segments merged
[ 2024/12/07 - 17:23:04 | pretty-gpx | INFO ] 0 segments merged
```

As you see the newer algorithm merge more and merge everything on the first pass.

For forest and rivers as I used before a reccursive merge with a max depth in merge_ways_closed_shapes. Now as the merge is perfect on the first time, it divides by 2 the time needed for rivers and forest (1.8s -> 0.85s).

For roads time is similar on 1 pass, but if we did multiple pass on roads then this update divides the time needed by the number of call to merge ways

Reply:
Looks great!